### ISSUE TITLE
implement color-blind friendly mode / light mode

### ISSUE THREAD

**Issue by CalAlaera** [dhowe/AdNauseam#2354]
### Describe the issue

AdNauseam does not have a light mode, which is needed for accessibility by users with certain vision problems. Please add one, ideally respecting the browser's light/dark mode settings.

### Screenshot in which the issue can be seen
Virtually any screen before or after install, including but not limited to:
![Screen1](https://github.com/dhowe/AdNauseam/assets/59891537/a5437c11-0d60-4663-8329-4238c1f20336)
![Screen2](https://github.com/dhowe/AdNauseam/assets/59891537/de21c281-9603-46ee-985e-e61a06419ae1)
![Website](https://github.com/dhowe/AdNauseam/assets/59891537/9c09a13e-c35c-4870-b3d1-d29fd137c82e)

### Steps for anyone to reproduce the issue
1. Set your browser to light mode
2. Install AdNauseam

### Your settings

- OS/version: Windows 11
- Browser/version: Firefox 115.0.2
- AdNauseam version: 3.17.0
- Other extensions you have installed: N/A
- AdNauseam settings: Default (first install)

##### Your filter lists
Default (first install)

### Your location/country
United Kingdom

**Comment by mneunomne** [comment]
Thats a good point @CalAlaera. I had considered implementing one before, but didn't manage to do so. I wonder if there is a simple way with a css overlay filter to do that.

**Comment by CalAlaera** [comment]
@mneunomne To be clear, I'm not a developer, and I know nothing about CSS, so please take this as the uninformed speculation that it is. But could it be as simple as just removing colours from the CSS and allowing the browser to use its defaults?

**Comment by ice-fly** [comment]
@CalAlaera I quickly went and inverted `common.css`, `menu.css`, and `dashboard.css` using https://jalu.ch/coding/css-negative.php
Although you could probably do a better job inverting the black and whites only using 5 minutes on chat gpt to retain the blue/ purple color scheme.
![image](https://github.com/dhowe/AdNauseam/assets/11548922/d3e59891-80af-4c76-a06f-aa271eada8fa)
![image](https://github.com/dhowe/AdNauseam/assets/11548922/9d82971a-dae2-408d-b69b-0da5d00d9433)
The real problem, as you can see is the tiled background `gray_grid.png`. Perhaps instead of using a tiled image we could switch to a CSS only method of making the grid?:
[How to make a grid (like graph paper grid) with just CSS?](https://stackoverflow.com/questions/3540194/how-to-make-a-grid-like-graph-paper-grid-with-just-css)
There are also images which are just logos on white background. Those would need to be added to the CSS/HTML as svgs: [Invert svg image using css?](https://stackoverflow.com/questions/39761120/invert-svg-image-using-css)

if you can recreate the images in css, then maybe another developer could implement a `@media (prefers-color-scheme: light)` to catch the cases where someone is running light mode on OS/browser. _(I assume you are doing that?)_

Try playing around with inspect element. Who knows maybe you'll become a big contributor?

**Comment by mneunomne** [comment]
Yes, a simple inversion of css colors won't do because we have multiple assets we use as well. Maybe the best (and most proper) solution would be to simply create a light theme on css and substitute all colors and have alternative light-mode assets.

**Comment by dhowe** [comment]
Lets make this happen for the next release

**Comment by mneunomne** [comment]
@dhowe Should we make a light theme, or a color-blind-friendly mode? The light implies that everything will be light-themed if the user browser configuration is also light. A color blind mode would only be applied of the user checks in the option on the dashboard "Color-blind friendly"

**Comment by dhowe** [comment]
my sense is that light/accessible mode should be opt-in only, via settings... this would imply the latter option above (unless the former requires less implementation time) -- note also that we already have a color-blind friendly option in settings

**Comment by mneunomne** [comment]
https://addons.mozilla.org/de/firefox/addon/adnauseam/reviews/1981205/

"Good idea, but completely unusable for me thanks to being locked in dark mode and not respecting my light-mode-only accessibility settings."

Should definitely go to next release.

**Comment by mneunomne** [comment]
Implemented light/color-blind-friendly mode. Not optimal aesthetics, but maybe good enough for next release. @dhowe what do you think?

I was able to finally refactor our css and now created a set adn-color-palette which will make it much more easy if we want to change our theme. There are still many improvements that can be done in this refactoring, but it is already much better now. @dhowe if you think it looks good then we merge, otherwise we can make some design readjustments, then do some thorough test to see if I didn't miss anything. The vault needs a bit if re-design in light-mode perhaps.

<img width="1440" alt="Screenshot 2023-10-06 at 01 43 33" src="https://github.com/dhowe/AdNauseam/assets/4967860/3cb189d4-a1e0-4baa-9a86-68ac12ddf081">
<img width="628" alt="Screenshot 2023-10-06 at 01 43 26" src="https://github.com/dhowe/AdNauseam/assets/4967860/4e724b53-2b55-4d24-a888-5779520fd0ca">
<img width="318" alt="Screenshot 2023-10-06 at 01 43 19" src="https://github.com/dhowe/AdNauseam/assets/4967860/a6234fe5-4b5b-45b8-ad09-edfaeff2b3ae">
<img width="317" alt="Screenshot 2023-10-06 at 01 43 12" src="https://github.com/dhowe/AdNauseam/assets/4967860/0c4a24dc-f6b2-4734-97a5-fe0ac8e5792c">
<img width="987" alt="Screenshot 2023-10-06 at 01 43 04" src="https://github.com/dhowe/AdNauseam/assets/4967860/c253ed12-020e-477d-9b50-55d05b183f13">
<img width="989" alt="Screenshot 2023-10-06 at 01 42 58" src="https://github.com/dhowe/AdNauseam/assets/4967860/993f6a7f-b707-4d49-b55e-517659ea4f71">
<img width="985" alt="Screenshot 2023-10-06 at 01 42 53" src="https://github.com/dhowe/AdNauseam/assets/4967860/74bfe59b-99cb-4714-9cbf-70c8b87ee7a4">
<img width="989" alt="Screenshot 2023-10-06 at 01 42 49" src="https://github.com/dhowe/AdNauseam/assets/4967860/a1ff2589-69a9-4448-85c3-e6e9a6087f6f">
<img width="980" alt="Screenshot 2023-10-06 at 01 42 45" src="https://github.com/dhowe/AdNauseam/assets/4967860/01b54524-61e5-46b5-87c2-1a62b89494db">
<img width="988" alt="Screenshot 2023-10-06 at 01 42 41" src="https://github.com/dhowe/AdNauseam/assets/4967860/86ba3370-27f0-41af-a9b7-2f797b9bc473">
<img width="983" alt="Screenshot 2023-10-05 at 23 26 56" src="https://github.com/dhowe/AdNauseam/assets/4967860/dc9ed586-4e05-43d4-acc2-335bdc1418e0">
<img width="1440" alt="Screenshot 2023-10-06 at 01 48 33" src="https://github.com/dhowe/AdNauseam/assets/4967860/c7b3b558-b05e-4a76-8aa2-b6defebd3717">

**Comment by dhowe** [comment]
This is a very good start, but we should get @mushon to weigh in

meantime, 2 questions:
1. Would it be it difficult to remove the background grid in this mode, and just have a pale gray color instead ?
2. I think we should avoid the use of white text on a very light bg if possible, such as below

<img width="444" alt="Screenshot 2023-10-06 at 6 05 43 PM" src="https://github.com/dhowe/AdNauseam/assets/737638/30fa0da8-8c7e-4179-9b34-6d5f30b19239">

**Comment by mushon** [comment]
generally I agree, but I'm not sure where this screenshot is coming from, the background is supposed to be dark like this:
<img width="419" alt="image" src="https://github.com/dhowe/AdNauseam/assets/27123/fc6a60a6-60c4-4954-abb4-3c83b5945bc6">

That makes both the white titles have high color contrast and the background pattern too subtle to be an issue for accessibility.

**Comment by dhowe** [comment]
that screenshot is from the colorblind mode above that @mneunomne has been working on (see [screenshots here](https://github.com/dhowe/AdNauseam/issues/2354#issuecomment-1749803380))

**Comment by mushon** [comment]
ok, re-reading the thread. I am not sure accessibility means mandatory light-mode, but if that's the way you want to go then obviously we will have to invert the light titles and there's no problem to remove the pattern as well.

**Comment by CalAlaera** [comment]
I'm very grateful my feature request has been taken seriously and people are working on it! Though just to clarify, I did specifically ask for a light mode, preferably respecting the browser's choice for light or dark mode.

**Comment by mneunomne** [comment]
@mushon it seems that couple users said they have issues using dark mode because of visual impairment, so that's we decided to enable in this way. From what I read there are multiple types of visual impairment and dark mode affects at least some forms of it.

**Comment by mneunomne** [comment]
And @mushon, maybe would be good if we have a visual reference on figma of the light mode? I did it on the go by changing to the colors we were already using on our interaface, but I am not really an interface designer.

**Comment by mushon** [comment]
Yeap, fair enough.
In that case we need to make sure all texts meet the minimum contrast requirements. I recommend not to use color inverse as the starting point but to actually work with a white background and use high contrast for text elements. Quick screenshot:

**Comment by mneunomne** [comment]
Already implement, specific issues regarding light-mode can be created in separate tickets. Closing now.
